### DISCUSSION TITLE
Strategies for resource management / allocation  when using `DinD` sidecar mode?

### DISCUSSION THREAD

**Discussion by snooyen** [actions/actions-runner-controller#3284] [Questions]
Wanted to see if anybody in the community has had an experience or tips with this particular subject.

For context, our organization is in the process of migrating our self-hosted GHA runners from the original `fairwinds` controller to the new RSS controller maintained by GitHub.  With the legacy controller, all of our RunnerDeployments are using the `-dind` runner images and we have distinct RunnerDeployment configurations based on CPU & Memory resource requests & limits which essentially match AWS EC2 T-series instance types.  That is to say, we have RunnerDeployments labeled `small`, `medium`, `large`, `xlarge` etc... where `medium` RunnerDeployment pods have resource requests an limits of 2 vCPU and 4 GiB Memory.  Since the runner pods are single container pods, this works well as the resource allocation covers both the runner software AND the Docker daemon.

It looks like the new RSS controller deploys pods using the vanilla / bare-bones runner container image published by GitHub and instead of offering a `DinD` variant, `DinD` capabilities are achieved by deployed the Docker daemon as a sidecar container.  So that leads me to the question: **for those using `DinD` sidecar mode, how do you manage resources on your runner pods?**

My initial thought might be that we could split up our RSS configurations into DinD and non-DinD runners.  For example, a `medium-DinD` runner would have 2vCPU and 4GiB of memory allocated to the DinD sidecar and then only a minimal amount of resource allocated to the runner container.  Conversely, the non-DinD container would have the full 2vCPU and 4GiB of memory allocated to the runner container.

**Comment by snooyen** [comment]
We ended up building our own runner images to support DinD within single-container pods.

**Reply by apdunn2** [reply]
I'm encountering the same problem. Is there a good third party image I can reference? Or did you do it from scratch?

**Reply by snooyen** [reply]
Wrote it from scratch!
